action/receipt.go

```diff
@@ -12,24 +12,67 @@ import (
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/golang/protobuf/proto"
 
+	"github.com/iotexproject/go-pkgs/byteutil"
 	"github.com/iotexproject/go-pkgs/hash"
+	"github.com/iotexproject/iotex-address/address"
 	"github.com/iotexproject/iotex-proto/golang/iotextypes"
 
 	"github.com/iotexproject/iotex-core/pkg/log"
 )
 
+// constants
+const (
+	StakingProtocolID   = "staking"
+	WithdrawAmountTopic = "withdrawAmount"
+)
+
 // InContractTransfer is topic for system log of evm transfer
 var InContractTransfer = common.Hash{} // 32 bytes with all zeros
 
 // IsSystemLog checks whether a log is system log
-// lowerBound chooses the largest system log topic, which is InContractTransfer currently
 func IsSystemLog(l *Log) bool {
-	if len(l.Topics) == 0 {
+	return IsEvmTransfer(l) || IsWithdrawBucket(l)
+}
+
+// IsWithdrawBucket checks withdraw bucket log
+func IsWithdrawBucket(l *Log) bool {
+	if l == nil || len(l.Topics) < 3 {
```

Comment:
if `l.Topics[0] == hash.BytesToHash256([]byte(WithdrawAmountTopic))` is true, `len(l.Topics) < 3` should be a fatal error

Reply:
i think it is a quick check, if topics < 3, we don't need to check further, it cannot be a withdraw bucket log